public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -194,12 +219,67 @@
 
     @Override
     public ResponseEntity<DbDomainListResponse> getDomainSearchResults(String keyword){
+
+        String[] keywords=keyword.split("[,+\\s+]");
+        for(int i=0;i<keywords.length;i++){
+            String key=keywords[i];
+            if(key.length() < 3){
+                key="\""+key+"\"";
+                keywords[i]=key;
+            }
+        }
+
+        String query2=new String();
+        for(String key:keywords){
+            if(query2.length()==0){
+                query2="+"+key;
+            }else if(key.contains("\"")){
+                query2=query2+key;
+            }else{
+                query2=query2+"+"+key;
+            }
+        }
+
+        keyword=query2;
+
         List<DbDomain> domains=dbDomainDao.findDomainSearchResults(keyword);
         DbDomainListResponse resp=new DbDomainListResponse();
         resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
         return ResponseEntity.ok(resp);
     }
 
+    @Override
+    public ResponseEntity<ConceptListResponse> getAdvancedConceptSearch(String keyword,String conceptFilter,List<String> vocabularyIds,List<String> domainIds){
+
+        Integer maxResults=25;
+        StandardConceptFilter standardConceptFilter;
+        if(conceptFilter == null){
+            standardConceptFilter = StandardConceptFilter.ALL_CONCEPTS;
+        }else if("S".equals(conceptFilter)){
+            standardConceptFilter = StandardConceptFilter.STANDARD_CONCEPTS;
+        }else if("C".equals(conceptFilter)){
```

Comment:
So my understanding is that "C" stands for "ancestor" and actually is used for standard concepts (for concepts that are only used for hierarchy, and shouldn't be used directly in the data). I have a TODO to change ConceptService so that it looks for both "S" and "C" when standard concepts are selected... though it may make more sense to do this once we have descendant counts in the data, since I believe all the "C" concepts right now will have zero counts.

Anyway, "C" shouldn't mean non-standard. I think you want this in your else clause instead.